### ISSUE TITLE
Visible focus confusing in closed widget areas

### ISSUE THREAD

**Issue by tellthemachines** [WordPress/gutenberg#38266]
### Description

When a widget area is closed, and in focus, the focus ring around it is indistinguishable from the focus ring around the widget area header. However, it's only possible to open the widget area with Enter or Space if the widget header is focused; not when focus is on the whole widget area. This makes it seem that Enter/Space keys randomly don't work when trying to open a widget area, because if focus is on the widget area, the keys do nothing, even though it looks like focus is on the header.

I'm not sure if it's best to try creating distinct visual focus styles for each state, as with the widget area closed, it'll always be hard to tell which is which. Perhaps allowing Enter and Space to open/close when the whole widget area is focused would be a better solution. It would also be good to consider the screen reader experience here.

### Step-by-step reproduction instructions

1. In the Widgets editor at `wp-admin/widgets.php`, focus the first widget area by tabbing from the top of the page
2. Press Arrow Down to focus the widget area header
3. Use Space or Enter to close the widget area
4. Toggle focus between header and whole widget area with Arrow Up and Arrow Down buttons
5. Notice there is almost no difference in visible focus when the widget area is closed
6. Try pressing Enter or Space when whole widget area is focused, note nothing happens

### Screenshots, screen recording, code snippet

Closed widget area (whole widget area focused):
<img width="735" alt="Widget area header with blue focus ring" src="https://user-images.githubusercontent.com/8096000/151290085-6d88c6df-e05d-4b61-9eff-75f0f6db86f0.png">

Closed widget area (widget area header focused):
<img width="726" alt="Widget area header with blue focus ring" src="https://user-images.githubusercontent.com/8096000/151290089-92bdff5d-6acd-4cb3-9b1e-4bea1f39dc77.png">

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by noisysocks** [comment]
Good catch. That's no good. Any examples of similar accordion UIs that get this right?

**Comment by JustinyAhin** [comment]
@noisysocks @annezazu maybe we can do something like in the menu editor here.

![CleanShot 2022-05-10 at 11 39 10@2x](https://user-images.githubusercontent.com/33403964/167599154-087a3769-8736-4935-83ec-ab5acc7b6c55.png)

That way, the focus will be only on the dropdown element and not on the whole header.
